contracts/forwarder/Eip712Forwarder.sol

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier:MIT
```

Comment:
General comment :
I don't see the benefit of braking down the forwarder in multiple inheritance contracts

We cannot expect receiver to choose which forwarder they support. The forwarder should be supporting features that we deemed necessary.

In my opinion, these are
- batch calls
- msg.value forwarding
- flexible nonce protection

And I am on the fence with :
- smart contract wallet support 

If the overhead is minimal (checking if sender is a smart contract)
I would go with it as it enable all smart contract wallet that support EIP-1271 and EIP-1654, to be supported by wallet that support GSN or other native metatx  enabledrelayer service